docs/source/developers/architecture.rst

```diff
@@ -0,0 +1,62 @@
+.. _architecture:
+
+Architecture Diagrams
+=====================
+
+This page describes the Jupyter Server architecture and the main workflows.
+This information is useful for developers who want to understand how Jupyter
+Server components are connected and how the principal workflows look like.
+
+To make changes for these diagrams, use `the Draw.io <https://app.diagrams.net/>`_
+open source tool.
+
+
+Jupyter Server Architecture
+---------------------------
+
+The Jupyter Server system can be seen in figure bellow:
+
+.. image:: ../images/jupyter-server-architecture.drawio.png
+   :alt: Jupyter Server Architecture
+   :width: 100%
+   :align: center
+
+Jupyter Server contains the following components:
+
+- **ServerApp** is the main Tornado-based application which connects all
+  components together.
+
+- **Config Manager** initializes configuration for the ServerApp. You can define
+  custom classes for the Jupyter Server managers using this config and change
+  SererApp settings. Follow :ref:`the Config File Guide <other-full-config>` to
+  learn about configuration settings and how to build custom config.
+
+- **Custom Extensions** allow to create the custom Server's REST API endpoint.
+  Follow :ref:`the Extension Guide <extensions>` to know more about extending
+  ServerApp with extra request handlers.
+
+- **Kernel Gateway** is the web server that provides access to Jupyter Kernels.
+  There are different ways to create this gateway. If your ServerApp needs to
+  communicate with remote Kernels, you can use
+  `the Enterprise Gateway <https://github.com/jupyter-server/enterprise_gateway>`_,
+  otherwise you can use `the Kernel Gateway <https://github.com/jupyter-server/kernel_gateway>`_.
```

Comment:
Because the box states, `Kernel Gateway` yet both forms of gateway servers are listed here, perhaps using the term `Gateway Server` may be more clear:
```suggestion
- **Gateway Server** is a web server that, when configured, provides access to Jupyter Kernels running on other hosts.
  There are different ways to create a gateway server. If your ServerApp needs to
  communicate with remote Kernels residing within resource-managed clusters, you can use
  `Enterprise Gateway <https://github.com/jupyter-server/enterprise_gateway>`_,
  otherwise, you can use `Kernel Gateway <https://github.com/jupyter-server/kernel_gateway>`_, where Kernels run locally to the gateway server.
```

Reply:
I agree with that, I will rename `Kernel Gateway` to `Gateway Server` on the diagram also.